### DISCUSSION TITLE
Is it possible to export scaleObjectFromCorner, scaleObjectX and scaleObjectY from controlsUtils?

### DISCUSSION THREAD

**Discussion by ritali4912** [fabricjs/fabric.js#9919] [Q&A]
I have a requirement that set minimal size of Rectangle, that means if the size reach to minimal, it can't scaling continue.
What in my mind is after the control actionHandler executed and reach to minimal size, I will set back the last scaleX, scaleY.

But current control actionHandler is wrapped, I need to wrap again if I set scaleX, scaleY back - wrapWithFixedAnchor, wrapWithFireEvent.

So if I can extend scaleObjectFromCorner, then I can simply replace
`
export const scalingEqually = wrapWithFireEvent(
  'scaling',
  wrapWithFixedAnchor(scaleObjectFromCorner)
);
`

**Or is there convenient way to implement the minimal size?**
Thanks so much.

**Comment by asturur** [comment]
Build additional wrapper is the way i would suggest.
In general scaling rectangles is not the best approach, if you are doing  things with rects, changing width and height is a bit easier to reason with.
Do you have something that is stopping you to create a wrapper 'withMinimalSize' or something like that? do you miss informations in the parameters?

**Comment by ritali4912** [comment]
@asturur 
Here is the jsfiddle, please have a look on function wrapExtendScalingAction:
https://jsfiddle.net/ritali4912/wbosyhk8/47/

I need to keep all default control behavior for `scaling`, but when reach the minimum size, it will stop scaling.

Because the `scaleObjectFromCorner`, `scaleObjectX` and `scaleObjectY` is internal, I need to wrapper my logic with `controlsUtils.scalingEqually` which will fire 'scaling' event and it is not what I want if < minimum size. So I think it is better to wrap my logic with `scaleObjectFromCorner`.

If there is easier way, it will very helpful.
Thanks so much.

**Reply by asturur** [reply]
Sorry for not answering before. This is a decent usecase, maybe we can see if some extra export is needed or what. I don't have time to look at it today. As a rough estimate the 'scaleObject' function needs to be split in calculateScale and applyScale so that you can use calculateScale in your function to know how much scale will be, and return false if you don't like it.
